src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1024,3 +1075,478 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
+            if column_type is None:
+                logger.error("Specified column does not exist in schema")
+                raise ValueError("Column does not exist")
+
+            if not isinstance(column_type, _NumericType) and not isinstance(
+                column_type, DateType
+            ):
+                logger.error(f"Unsupported column type: {column_type}")
+                raise ValueError(f"unsupported type {column_type}")
+            partitioned_queries = self._generate_partition(
+                select_query,
+                column_type,
+                column,
+                lower_bound,
+                upper_bound,
+                num_partitions,
+            )
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            # create temp table
+            snowflake_table_type = "temporary"
+            snowflake_table_name = random_name_for_temp_object(TempObjectType.TABLE)
+            create_table_sql = (
+                "CREATE "
+                f"{get_temp_type_for_object(self._session._use_scoped_temp_objects, True) if snowflake_table_type.lower() in TEMPORARY_STRING_SET else snowflake_table_type} "
+                "TABLE "
+                f"{snowflake_table_name} "
+                f"""({" , ".join([f'"{field.name}" {convert_sp_to_sf_type(field.datatype)} {"NOT NULL" if not field.nullable else ""}' for field in struct_schema.fields])})"""
+                f"""{DATA_SOURCE_SQL_COMMENT}"""
+            )
+            logger.debug(f"Creating temporary Snowflake table: {snowflake_table_name}")
+            self._session.sql(create_table_sql).collect(
+                statement_params=statements_params_for_telemetry
+            )
+            # create temp stage
+            snowflake_stage_name = random_name_for_temp_object(TempObjectType.STAGE)
+            sql_create_temp_stage = (
+                f"create {get_temp_type_for_object(self._session._use_scoped_temp_objects, True)} stage"
+                f" if not exists {snowflake_stage_name} {DATA_SOURCE_SQL_COMMENT}"
+            )
+            self._session.sql(sql_create_temp_stage).collect(
+                statement_params=statements_params_for_telemetry
+            )
+
+            try:
+                with mp.Manager() as process_manager, ProcessPoolExecutor(
+                    max_workers=max_workers
+                ) as process_executor, ThreadPoolExecutor(
+                    max_workers=max_workers
+                ) as thread_executor:
+                    thread_pool_futures, process_pool_futures = [], []
+                    parquet_file_queue, process_error_queue = (
+                        process_manager.Queue(),
+                        process_manager.Queue(),
+                    )
+
+                    def fetch_process_error_handling_callback(fetch_process_future):
+                        if fetch_process_future.exception():
+                            process_error_queue.put(fetch_process_future.exception())
```

Comment:
need to remove process_error_queue & fetch_process_error_handling_callback because they are useless, our error check relies directly on the result